src/Analyser/Scope.php

```diff
@@ -341,15 +344,62 @@ private function resolveType(Expr $node): Type
 			return new BooleanType();
 		}
 
-		if (
-			$node instanceof Node\Expr\UnaryMinus
-			|| $node instanceof Node\Expr\UnaryPlus
+		if ($node instanceof Node\Expr\UnaryPlus
 			|| $node instanceof Expr\ErrorSuppress
 			|| $node instanceof Expr\Assign
 		) {
 			return $this->getType($node->expr);
 		}
 
+		if ($node instanceof Node\Expr\UnaryMinus) {
+			$type = $this->getType($node->expr);
+			if ($type instanceof ConstantIntegerType) {
+				return new ConstantIntegerType(-$type->getValue());
+			} elseif ($type instanceof ConstantFloatType) {
+				return new ConstantFloatType(-$type->getValue());
+			} else {
+				return $type;
+			}
+		}
+
+		if ($node instanceof Node\Expr\BinaryOp || $node instanceof Node\Expr\AssignOp) {
+			if ($node instanceof Node\Expr\AssignOp) {
+				$left = $node->var;
+				$right = $node->expr;
+			} elseif ($node instanceof Node\Expr\BinaryOp) {
+				$left = $node->left;
+				$right = $node->right;
+			} else {
+				throw new \PHPStan\ShouldNotHappenException();
+			}
+
+			$leftType = $this->getType($left);
+			$rightType = $this->getType($right);
+
+			if ($leftType instanceof ConstantScalarType && $rightType instanceof ConstantScalarType) {
+				$leftValue = $leftType->getValue();
+				$rightValue = $rightType->getValue();
+
+				if ($node instanceof Node\Expr\BinaryOp\Plus || $node instanceof Node\Expr\AssignOp\Plus) {
+					return @$this->getTypeFromValue($leftValue + $rightValue);
```

Comment:
error suppress operator? why? 😱

Reply:
Because for example `'A' + 'B'` emits warning „PHP warning:  A non-numeric value encountered“ and I this was the easiest solution.

Reply:
OK, so it would be better to do something like:

```php
$result = @($leftValue + $rightValue);
return $this->getTypeFromValue($result);